### PR TITLE
[MERGE][FIX] knowledge: fix "move to private" and archive flows

### PR DESCRIPTION
**Pull request by awa-odoo**
Fixes several issues into knowledge flows:
- Correctly test that the article still has a member when moving to a private
  section
- Copy member values when detaching unreachable children during archiving
- Propagate to children when making an article 'private'

See underlying commits for details.

Task-2859616

### CODE REVIEW

File: addons/knowledge/models/knowledge_article.py [hunk 1]
```diff
@@ -760,15 +750,24 @@ def move_to(self, parent_id=False, before_article_id=False, is_private=False):
                 )
             self_sudo = self.sudo()
             self_member = self_sudo.article_member_ids.filtered(lambda m: m.partner_id == self.env.user.partner_id)
-            member_command = [(2, member.id) for member in self_sudo.article_member_ids if member.partner_id != self.env.user.partner_id]
+
             if self_member:
-                member_command.append((1, self_member.id, {'permission': 'write'}))
+                write_member_command = [(1, self_member.id, {'permission': 'write'})]
```
Comment (abd-msyukyu-odoo) [inline addons/knowledge/models/knowledge_article.py:755]: Shouldn't we use `Command.update` for better readability? (also for subsequent commands)

File: addons/knowledge/models/knowledge_article.py [hunk 2]
```diff
@@ -1073,6 +1072,34 @@ def _remove_member(self, member):
             if current_membership:
                 self.sudo().write({'article_member_ids': [(2, current_membership.id)]})
 
+    def _has_write_member(self, child_members=False, exclude_memberships=False):
+        """ Method allowing to check if this article still have at least one member with write access.
```
Comment (abd-msyukyu-odoo) [inline addons/knowledge/models/knowledge_article.py:1076]: has*

**Review by abd-msyukyu-odoo — COMMENTED** [review]
Small suggestions ;p

File: addons/knowledge/models/knowledge_article.py (hunk 1 above)
Comment (awa-odoo) [reply addons/knowledge/models/knowledge_article.py:755]: I see that the knowledge code base uses mostly the "magic numbers" writing.
(`_inverse_is_user_favorite`, `_remove_member`, ...).

Maybe it's best to keep it like that for consistency.

File: addons/knowledge/models/knowledge_article.py (hunk 2 above)
Comment (awa-odoo) [reply addons/knowledge/models/knowledge_article.py]: Good catch! It's fixed :smile:

File: addons/knowledge/models/knowledge_article.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/knowledge/models/knowledge_article.py:755]: Maybe because "someone" did go through the python code and still does not use the commands.

**Comment by C3POdoo** [comment]
Dear @awa-odoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**